app/src/main/java/com/practicum/playlistmaker/AudioPlayerActivity.kt

```diff
@@ -70,28 +108,90 @@ class AudioPlayerActivity : AppCompatActivity() {
         }
     }
 
+    private fun startPlayer() {
+        if (mediaPlayer == null) {
+            Toast.makeText(this, "MediaPlayer не готов", Toast.LENGTH_SHORT).show()
+            return
+        }
+        mediaPlayer?.start()
+        isPlaying = true
+        playButton.setImageResource(R.drawable.btn_aud_pause)
+        startUpdatingTime()
+    }
+
+    private fun pausePlayer() {
+        mediaPlayer?.pause()
+        isPlaying = false
+        playButton.setImageResource(R.drawable.btn_aud_play)
+        stopUpdatingTime()
+    }
+
+    private fun stopPlayer() {
+        mediaPlayer?.seekTo(0)
+        mediaPlayer?.pause()
+        isPlaying = false
+        playButton.setImageResource(R.drawable.btn_aud_play)
+        currentTrackTime.text = "00:00"
+        stopUpdatingTime()
+    }
+
+    private fun startUpdatingTime() {
+        updateTimeRunnable = object : Runnable {
+            override fun run() {
+                val currentPosition = mediaPlayer?.currentPosition ?: 0
+                Log.d("AudioPlayer", "Updating time: $currentPosition ms")
+                currentTrackTime.text = formatTime(currentPosition)
+                handler.postDelayed(this, 500)
+            }
+        }
+        handler.post(updateTimeRunnable!!)
```

Comment:
Чтобы избавиться от использования оператора `!!`:
```suggestion
        updateTimeRunnable = object : Runnable {
            override fun run() {
                val currentPosition = mediaPlayer?.currentPosition ?: 0
                Log.d("AudioPlayer", "Updating time: $currentPosition ms")
                currentTrackTime.text = formatTime(currentPosition)
                handler.postDelayed(this, 500)
            }
        }.let { handler.post(it) }
```